src/Exception/AbstractBaseTokenizerException.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Dhii\Parser\Tokenizer\Exception;
+
+use RuntimeException;
+use Dhii\Parser\Tokenizer\TokenizerAwareTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use Dhii\Util\Normalization\NormalizeIntCapableTrait;
+
+/**
+ * Common functionality for tokenizer exceptions.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractBaseTokenizerException extends RuntimeException implements TokenizerExceptionInterface
+{
+    /*
+     * Adds internal tokenizer awareness.
+     *
+     * @since [*next-version*]
+     */
+    use TokenizerAwareTrait;
+
+    /*
+     * Adds string normalization functionality.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeStringCapableTrait;
```

Comment:
The `NormalizeStringCapableTrait` and `NormalizeIntCapableTrait` are not used in this class. But they _are_ used in the derived concrete classes. 

On the other hand, the `CreateInvalidArgumentExceptionCapableTrait` and the `StringTranslatingTrait` are not here, but both derived concrete classes `use` them.

Consider either moving the latter 2 traits to this base class or to former 2 traits to the derived concrete classes. I recommend the latter since, so as to not include unused functionality in the base class that would then be included in derived classes.

Reply:
Will move everything to base class.